build.rs

```diff
@@ -72,6 +73,10 @@ fn main() {
     if rustc_dep_of_std {
         println!("cargo:rustc-cfg=libc_thread_local");
     }
+
+    if const_extern_fn_cargo_feature {
+        println!("cargo:rustc-cfg=libc_const_extern_fn");
```

Comment:
So here we should test whether the compiler is a new enough nightly compiler, and if not, `panic!("const-extern-fn requires a nightly compiler >= 1.40.0")` or so.

Reply:
Added